server/game/cards/04.4-TEaF/SoulBeyondReproach.js

```diff
@@ -1,10 +1,22 @@
 const DrawCard = require('../../drawcard.js');
 
 class SoulBeyondReproach extends DrawCard {
-    setupCardAbilities(ability) { // eslint-disable-line no-unused-vars
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Honor a character, then honor it again',
+            target: {
+                cardType: 'character',
+                controller: 'self',
+                gameAction: [
+                    ability.actions.honor(),
+                    ability.actions.honor()
```

Comment:
The way you've done this, it will apply two simultaneous honor effects to the target, whereas the card says 'Then, ...' so we really want them to be sequential.  The way to do this is to use a `then:` property - look at Test of Courage for an example.

The reason why it needs to be sequential rather than simultaneous is due to effects like Young Rumormonger.  If you young rumormonger the first honor effect, in your implementation the second will still happen, whereas it shouldn't as the first one didn't resolve successfully.